Tutorials/WinForms/Classic/create-persistent-classes-and-connect-xpo-to-database.md

```diff
@@ -0,0 +1,105 @@
+# Create persistent classes and connect XPO to the database
+
+&nbsp;&nbsp;&nbsp;
+[Back to TOC](../../)
+&nbsp;&nbsp;&nbsp;
+[Step 2](/connect-data-grid-to-xpo-objects.md)   
+
+* Create a new Windows Forms App project in Visual Studio.
+* Add references to the following assemblies ([Manage references in a project](https://docs.microsoft.com/en-us/visualstudio/ide/managing-references-in-a-project)):  
+  **DevExpress.Data**  
+  **DevExpress.Xpo**  
+  These assemblies are available under the **Assemblies > Extensions** category in the **Reference Manager**. 
+* Create the **DataAccess** folder and add the `Customer` class to it. This class is a model for the Customer table in a database.
+* Change the base class to `XPObject` and add a constructor with a `session` parameter:  
+    ``` csharp
+    using DevExpress.Xpo;
+
+    namespace DxSample.DataAccess {
+        public class Customer :XPObject {
+            public Customer(Session session) : base(session) { }
+        }
+    }
+    ```
+    **See also:**   
+    [XPO Classes Comparison](https://docs.devexpress.com/XPO/3311/concepts/xpo-classes-comparison)
+* Add the `FirstName` and `LastName` properties to the `Customer` class. XPO maps these properties to columns in the Customer table.
+    ```csharp
+    public string FirstName {
+        get { return fFirstName; }
+        set { SetPropertyValue(nameof(FirstName), ref fFirstName, value); }
+    }
+
+    private string fLastName;
+    public string LastName {
+        get { return fLastName; }
+        set { SetPropertyValue(nameof(LastName), ref fLastName, value); }
+    }
+    ```
+    The `SetPropertyValue` method raises the `PropertyChanged` event that is used by data-bound controls to update their display text.
+* Add the read-only `ContactName` property. XPO does not map read-only properties to database columns. Use the `FirstName` and `LastName` properties to calculate the `ContactName` property value.
+    ```csharp
+    public string ContactName {
+        get { return string.Concat(FirstName, " ", LastName); }
+    }
+    ```
+* Add the [PersistentAlias](https://docs.devexpress.com/XPO/DevExpress.Xpo.PersistentAliasAttribute) attribute to the `ContactName` property. 
+  >The [PersistentAlias](https://docs.devexpress.com/XPO/DevExpress.Xpo.PersistentAliasAttribute) attribute specifies an expression that substitutes for the property name in a SQL query. The expression passed to the attribute constructor should follow the [Criteria Language Syntax](https://docs.devexpress.com/CoreLibraries/4928/devexpress-data-library/criteria-language-syntax). 
```

Comment:
"**that substitutes for**"? Substitutes what for it?

Reply:
"should follow"? Do you mean 'should use'?

Reply:
>"that substitutes for"? Substitutes what for it?

Changed to " an expression that XPO should use instead of the property name"

>"should follow"? Do you mean 'should use'?

Should conform to or should be compatible with. I changed this to "Use Criteria Language to build the expression.

The final variant is: "The [PersistentAlias](https://docs.devexpress.com/XPO/DevExpress.Xpo.PersistentAliasAttribute) attribute specifies an expression that XPO should use instead of the property name in a SQL query. Use [Criteria Language](https://docs.devexpress.com/CoreLibraries/4928/devexpress-data-library/criteria-language-syntax) to build the expression."

bcda9cf74119c0520619f07cec1f14e0822d0825